### ISSUE TITLE
Flutter add-to-app instructions for Android have a bug for a11y users

### ISSUE THREAD

**Issue by dnfield** [flutter/website#3346]
https://add-to-app-2019.firebaseapp.com/docs/development/add-to-app/android/add-flutter-fragment - specifically, the `onBackPressed` doesn't fire when a TalkBack user does the "back" a11y gesture (swipe down and left).

/cc @matthew-carroll @xster

**Comment by dnfield** [comment]
Realizing my top level comment is missing deetails...  

If the fragment isn't removed, adding it again results in an exception. We need to have a better way to make sure it's removed, catching the onBackPressed doesn't work for TalkBack users.

**Comment by dnfield** [comment]
It works locally if we just check for the fragment before showing it and removing it. Not sure if that's really the best way to implement it.

**Comment by matthew-carroll** [comment]
@dnfield when you refer to `onBackPressed` are you referring to the `Activity` method?

Also, can you describe the expected behavior that we need to implement, which is currently broken?

**Comment by dnfield** [comment]
> onBackPressed are you referring to the Activity method?

Yes.

> Also, can you describe the expected behavior that we need to implement, which is currently broken?

The fragment is removed `onBackPressed`.  However, if you're using TalkBack and use the `Swipe down and then left` to go back (since otherwise pressing the back button is hard, especially on Android Q), it doesn't seem to trigger `onBackPressed`, which means the fragment doesn't get removed, which means when you click the button again we try to add a fragment that has already been added and crash.

**Comment by matthew-carroll** [comment]
> ...when you click the button again...

Looking at the linked guide I don't think I see any examples with button presses. Can you point me to the code in question?

**Comment by dnfield** [comment]
Hmm.  I think I might be crossing some wires here.

I really thought I saw a page last night that had code very similar to the one posted for https://github.com/flutter/flutter/issues/46029

But now I can't find it.  It's entirely possible that I just looked at the wrong tab at one point and thought I saw something that wasn't there.

And now, even with the bad version from last night, I'm finding it difficult to reproduce the issue I have - TalkBack is triggering onBackPressed now.

So I give up.  Let's close this.
